include/openssl/bio.h

```diff
@@ -452,6 +447,23 @@ OPENSSL_EXPORT int BIO_set_mem_buf(BIO *bio, BUF_MEM *b, int take_ownership);
 OPENSSL_EXPORT int BIO_set_mem_eof_return(BIO *bio, int eof_value);
 
 
+// BIO close flags.
+//
+// These can be used as symbolic arguments when a "close flag" is passed to a
+// BIO function.
+
+// BIO_NOCLOSE will not close the underlying file on BIO free
+#define BIO_NOCLOSE 0
+
+// BIO_CLOSE will close the underlying file on BIO free
+#define BIO_CLOSE 1
+
+// BIO_FP_TEXT will cause the file to be opened as a text file (this is only
+// relevant on Windows due to CRLF endings) instead of the default behavior of
+// opening the file in binary mode.
+#define BIO_FP_TEXT 0x10
```

Comment:
> Is there any situation where this flag is in effect when the file is "opened"? (The documentation here only addresses its use when opening a file.) I see it can be passed as part of close_flag for BIO_set_fp and BIO_new_fp, but in those cases the file has already been "opened", we just potentially change its mode.

I think my language is imprecise here. No file is ever actually `open`ed with this flag; it's set after open (although some functions immediately set it after opening before returning to the caller). The functions available to callers that support this flag are:

- `BIO_new_file`
- `BIO_new_fp`
- `BIO_set_fp`

I'll update the doc comments of those functions to reflect this and modify this doc comment to be more precise.

> Should this flag also be called out in the documentation for BIO_set_fp and BIO_new_fp?

Yes.